src/api/projectService.tsx

```diff
@@ -1,33 +1,103 @@
 /*
   This program has been developed by students from the bachelor Computer Science at Utrecht
-  University within the Software Project course.
+  University within the Software Models course.
 
   © Copyright Utrecht University (Department of Information and Computing Sciences)
 */
 
-import {queryOptions} from "@tanstack/react-query"
+import {UseMutationOptions, UseQueryOptions} from "@tanstack/react-query"
 import config from "../../config.tsx"
+import {Project} from "@/types/project.ts";
 
 /**
  * Creates a project query to be executed in the project page.
  * @param projectId the string id of the project
  */
-export const projectQuery = (projectId: string) => {
-    return queryOptions({
+export function projectQuery(projectId: string): UseQueryOptions<Project> {
+    return {
         queryKey: ["project", projectId],
-        queryFn: () => getProjectById(projectId)
-    })
-};
+        queryFn: (): Promise<Project> => getProjectById(projectId)
+    }
+}
+
+export function editProjectQuery(projectId: string, origional?: Project): UseMutationOptions<Project, Error, Project> {
+    return {
+        mutationFn: (update) => {
+
+            if (!origional) {
```

Comment:
fix spelling van original

```suggestion
export function editProjectQuery(projectId: string, original?: Project): UseMutationOptions<Project, Error, Project> {
    return {
        mutationFn: (update) => {

            if (!original) {
```